src/compiler/emitter.ts

```diff
@@ -1488,11 +1488,6 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
             }
 
             function emitExpressionIdentifier(node: Identifier) {
-                if (resolver.getNodeCheckFlags(node) & NodeCheckFlags.LexicalArguments) {

```

Comment:
See https://github.com/Microsoft/TypeScript/pull/5870#discussion_r50786034. I found I didn't need this if I switched `__awaiter` from issuing a `generator.call` to a `generator.apply`.